Support arm64 coroutines.
Don't downgrade arm64 branch protection.

Set correct shared library name.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Index: configure
--- configure.orig
+++ configure
@@ -10904,7 +10904,7 @@ esac
     case "$target_cpu" in #(
   aarch64) :
 
-	for opt in -mbranch-protection=pac-ret -msign-return-address=all
+	for opt in -mbranch-protection=standard -msign-return-address=all
 do :
 
 
@@ -29482,6 +29482,10 @@ esac
 
             coroutine_type=x86
          ;; #(
+  aarch64-openbsd*) :
+
+            coroutine_type=arm64
+         ;; #(
   *-openbsd*) :
 
             coroutine_type=pthread
@@ -31691,7 +31695,7 @@ fi
 	 ;; #(
   openbsd*|mirbsd*) :
 
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby32_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -33682,7 +33686,7 @@ then :
 
 else $as_nop
 
-        arch="${target_cpu}-${target_os}"
+        arch="${target_cpu}-openbsd"
 
 fi
     printf "%s\n" "#define RUBY_PLATFORM \"$arch\"" >>confdefs.h
